source/libNsdl/src/sn_grs.c

```diff
@@ -601,8 +603,18 @@ extern int8_t sn_grs_process_coap(struct nsdl_s *nsdl_handle, sn_coap_hdr_s *coa
 
                 memcpy(response_message_hdr_ptr->payload_ptr, resource_temp_ptr->resource, response_message_hdr_ptr->payload_len);
             }
+            // Add max-age attribute for static resources
+            if (static_get_request) {
+                response_message_hdr_ptr->options_list_ptr = handle->sn_grs_alloc(sizeof(sn_coap_options_list_s));
+                if (response_message_hdr_ptr->options_list_ptr) {
+                    response_message_hdr_ptr->options_list_ptr->max_age_ptr = handle->sn_grs_alloc(1);
+                    if (response_message_hdr_ptr->options_list_ptr->max_age_ptr) {
+                        response_message_hdr_ptr->options_list_ptr->max_age_ptr[0] = 0;
+                        response_message_hdr_ptr->options_list_ptr->max_age_len = 1;
+                    }
+                }
+            }
         }
```

Comment:
Most paths in this function seem to abort on out of memory case and return SN_NSDL_FAILURE. If this case is special and does not need that, please add some comment about it.